### PR TITLE
remove result_mapper for zip_with_iterable operator

### PR DESCRIPTION
**Pull request by jcafhe**
Remove result_mapper argument for zip operator. 

The resulting elements are now zipped in tuple.

### CODE REVIEW

**Comment by dbrattli** [comment]
I really like this, returning a tuple instead of using the result_mapper, and then composing with map afterwards. It is btw exactly how I do it in the  implementation of [zipSeq](https://github.com/dbrattli/Reaction/blob/master/AsyncRx/src/Combine.fs#L243) in Reaction.AsyncRx (F#). We should btw do the same with `combine_latest` and `with_latest_from` (imo).

File: rx/core/operators/zip.py [hunk 1]
```diff
@@ -61,7 +57,8 @@ def on_next(left):
                     right = second[index]
                     index += 1
                     try:
-                        result = result_mapper(left, right)
+                        result = (left, right)
+
                     except Exception as ex:
```
Comment (dbrattli) [inline rx/core/operators/zip.py]: Just remove the try/except as well. Not needed anymore.

File: rx/operators/__init__.py [hunk 2]
```diff
@@ -2550,32 +2550,25 @@ def zip(*args: Observable) -> Callable[[Observable], Observable]:
     return _zip(*args)
 
 
-def zip_with_iterable(second, result_mapper):
+def zip_with_iterable(second):
     """Merges the specified observable sequence and list into one
-    observable sequence by using the mapper function whenever all of
+    observable sequence by using a tupple aggregation whenever all of
```
Comment (dbrattli) [inline rx/operators/__init__.py]: by creating a tuple whenever all of ...

File: rx/operators/__init__.py [hunk 3]
```diff
@@ -2550,32 +2550,25 @@ def zip(*args: Observable) -> Callable[[Observable], Observable]:
...
-        result_mapper: Mapper function that produces an element
-            whenever all of the observable sequences have produced an
-            element at a corresponding index
 
     Returns:
         An operator function that takes and observable source and
         returns an observable sequence containing the result of
-        combining elements of the sources using the specified result
-        mapper function.
+        combining elements of the sources as tuple.
```
Comment (dbrattli) [inline rx/operators/__init__.py]: as a tuple

File: tests/test_observable/test_zip.py [hunk 4]
```diff
@@ -365,9 +365,10 @@ def test_zip_with_iterable_never_empty(self):
         n2 = []
 
         def create():
-            def mapper(x, y):
-                return x + y
-            return n1.pipe(ops.zip_with_iterable(n2, result_mapper=mapper))
+            return n1.pipe(
+                    ops.zip_with_iterable(n2),
+                    ops.map(sum),
```
Comment (dbrattli) [inline tests/test_observable/test_zip.py]: Indent not correctly (should be 4 spaces)

File: tests/test_observable/test_zip.py [hunk 5]
```diff
@@ -380,9 +381,10 @@ def test_zip_with_iterable_empty_empty(self):
...
+                    )
```
Comment (dbrattli) [inline tests/test_observable/test_zip.py]: Indent not correctly (should be 4 spaces)

File: rx/core/operators/zip.py (hunk 1 above)
Comment (jcafhe) [reply rx/core/operators/zip.py]: Nice catch, we should do the same for the zip operator.

File: tests/test_observable/test_zip.py (hunk 4 above)
Comment (jcafhe) [reply tests/test_observable/test_zip.py]: this should be fixed now.

**Comment by jcafhe** [comment]
I agree, returning tuples from `combine_latest`& `with_latest_from` would be great. So you would remove the mapper function from these operators?

The only drawback I see, if `combine_latest` returns a tuple and is piped with `map`, is that you can't provide anymore a function with positional arguments. You need to unpack the tuple in the function body.

**Comment by dbrattli** [comment]
Yes, I see the problem. Perhaps you would like to implement a new operator called `starmap`?  Inspired by https://docs.python.org/2/library/itertools.html#itertools.starmap

**Pull request merged**